frontend/src/hooks/useLookupCodes.tsx

```diff
@@ -13,7 +14,18 @@ function useCodeLookups() {
     return lookupCodes.filter(code => code.type === type && code.id === id)?.find(x => x)?.code;
   };
 
-  const getByType = (type: string) => lookupCodes.filter(code => code.type === type);
+  const getByType = useCallback(
```

Comment:
I think this is a good use of `useCallback`.  

One thing I don't fully understand related to this though is whether the results are also cached in memory, or is it just the function?  If it's just the function, I wonder how valuable `useCallback` is.

https://dmitripavlutin.com/dont-overuse-react-usecallback/

Reply:
I believe that we are in this situation here (from your article):

**However, there are cases when you need to keep one instance of a function:

A component wrapped inside React.memo() (or shouldComponentUpdate) accepts a callback prop**

ManageUsersPage has a useMemo which accepts getByType as a callback prop.

I don't think ManageUsersPage really requires the memoized agencies/roles. However, that was how that code was written prior to my change, and I was fine leaving it as-is